src/Symfony/Bridge/Doctrine/Types/DatePointType.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Types;
+
+use DateTimeInterface;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\ConversionException;
+use Doctrine\DBAL\Types\DateTimeImmutableType;
+use Symfony\Component\Clock\DatePoint;
+
+final class DatePointType extends DateTimeImmutableType
+{
+    public const NAME = 'date_point';
+
+    public function convertToPHPValue(mixed $value, AbstractPlatform $platform): ?DatePoint
+    {
+        if (null === $value || $value instanceof DatePoint) {
```

Comment:
```suggestion
        if (null === $value) {
```

Pretty sure that the database won't send us a `DatePoint` object.

Reply:
The proposed check is consistent with other Doctrine types.

Reply:
… which means that the upstream type has that check already. I'll open another comment, give me a sec. 🤔